CMakeLists.txt

```
@@ -282,6 +282,16 @@ if(ENABLE_CURL)
   endif()
 endif()
 
+################################################################################
+# Check for Emscripten fetch
+################################################################################
+
+option(ENABLE_EMSCRIPTEN_FETCH "Enable Emscripten fetch support" OFF)
+set(EMSCRIPTEN_FETCH_ENABLED FALSE)
+if(ENABLE_EMSCRIPTEN_FETCH)
+  set(EMSCRIPTEN_FETCH_ENABLED TRUE)
+endif()
```

why not turning it ON by default, but only exposing it if (EMSCRIPTEN) ?
```suggestion
if (EMSCRIPTEN)
  option(ENABLE_EMSCRIPTEN_FETCH "Enable Emscripten fetch support" ON)
  set(EMSCRIPTEN_FETCH_ENABLED FALSE)
  if(ENABLE_EMSCRIPTEN_FETCH)
    set(EMSCRIPTEN_FETCH_ENABLED TRUE)
  endif()
endif()
```

---

I see your point. But it may have some side effects. At the be beginning of networkfilemanager.cpp I check it with the macro  `__EMSCRIPTEN_PTHREADS__` .
Some users (a majority?) will prefer to build without the fetch functionality. That produces a smaller wasm file, without libtiff, zlib, -pthreads, and it works in the main thread. A Web Worker is not needed.
(well, compiling with pthreads also works in the main thread... until it does a fetch request ;)
In this case I think opt-in is easier and safer. Maybe should we say something else in the documentation? In https://proj.org/en/stable/usage/network.html ?

---

> At the be beginning of networkfilemanager.cpp I check it with the macro `__EMSCRIPTEN_PTHREADS__` .
> Some users (a majority?) will prefer to build without the fetch functionality.

ok, I see your point. Let's forget that then